### PR TITLE
clean and bugfix

### PR DESCRIPTION
**Pull request by mp3butcher**

### CODE REVIEW

File: src/osgAnimation/MorphTransformHardware.cpp [hunk 1]
```diff
@@ -46,23 +48,22 @@ bool MorphTransformHardware::init(MorphGeometry& morphGeometry)
     ///check for correct morph configuration
     ///(blender osgexport doesn't set sources so assume morphgeom arrays are sources)
     if(pos)
-    {
+    {   pos->setDataVariance(osg::Object::STATIC);
```
Comment (openscenegraph) [inline src/osgAnimation/MorphTransformHardware.cpp]: Could you put lines of code on the line following the { to keep the code readable and consistent with the rest of the OSG.

File: src/osgAnimation/RigTransformHardware.cpp [hunk 2]
```diff
@@ -333,11 +347,10 @@ bool RigTransformHardware::init(RigGeometry& rig){
         stateset->addUniform(_uniformMatrixPalette);
 
         stateset->removeAttribute(osg::StateAttribute::PROGRAM);
-        if(!stateset->getAttribute(osg::StateAttribute::PROGRAM))
-            stateset->setAttributeAndModes(program.get());
+        stateset->setAttributeAndModes(program.get());
```
Comment (openscenegraph) [inline src/osgAnimation/RigTransformHardware.cpp]: The removeAttribute could removed as the setAttributes() will be replace the previous one automatically.  The setAttributeAndModes() could be replaced with setAttribute() as there aren't any GL modes associated with osg/glProgram,

**Comment by openscenegraph** [comment]
@mp3butcher I've down a quick run thriough and made a few comments of minor items hat could be improved. Could you resolve these then I'll merge. Cheers.

**Comment by mp3butcher** [comment]
@robertosfield  done 
Let me know if you see other stuff to clean 
I have review this 3 time but I still miss few development artifacts...

**Comment by mp3butcher** [comment]
@robertosfield 
please review my last commit: I don't know if it's a good one for readability

**Comment by openscenegraph** [comment]
@mp3butcher I did another review and it looks like things are OK now, but I'm not familiar with the code at a lower level so can't comment on the validity of changes in function by just looking at github code changes.  

I did attempt a merge but got an error about the Head of the branch having been modified and suggesting reviewing the changes, but don't know what might be amiss as I haven't personally made any changes to the branch, only merged PR's from yourself.

Travis is still running a check so I'll wait for this to complete before attempting another merge.

**Pull request merged**